### ISSUE TITLE
Detecting Ramses cosmological runs without namelist.txt

### ISSUE THREAD

**Issue by Martin-Rey** [pynbody/pynbody#597]
Hi @mkrets  and @apontzen,
 
Since #534 and #582, the detection of cosmological runs with Ramses relies exclusively on opening and reading the namelist:
https://github.com/pynbody/pynbody/blob/d9124e7b76d6f768e707d1aee0c45d2303ce9162/pynbody/snapshot/ramses.py#L996-L1000
when it previously relied on cosmological parameter information. While reading the namelist is a technical improvement, it happens quite often that the namelist.txt, patches.txt or compilation.txt files from Ramses get corrupted on supercomputers, and are unreadable. In such a case, pynbody will throw the following user warning
```
/cosma6/data/dp171/shared/python/anaconda3/lib/python3.8/site-packages/pynbody-1.0.2-py3.8-linux-x86_64.egg/pynbody/snapshot/ramses.py:537: UserWarning: Namelist found but unable to read.
  warnings.warn("Namelist found but unable to read.")
```
and conclude that the run is not cosmological. This leads to negative simulation ages, even if cosmogical parameters are read in and one can compute the correct `age' with analysis routines
```
In [1]: import pynbody                                                                                                                                                                                                    

In [2]: s = pynbody.load("../Halo383_fiducial/output_00060")                                                                                                                                                              
/cosma6/data/dp171/shared/python/anaconda3/lib/python3.8/site-packages/pynbody-1.0.2-py3.8-linux-x86_64.egg/pynbody/snapshot/ramses.py:537: UserWarning: Namelist found but unable to read.
  warnings.warn("Namelist found but unable to read.")
/cosma6/data/dp171/shared/python/anaconda3/lib/python3.8/site-packages/pynbody-1.0.2-py3.8-linux-x86_64.egg/pynbody/snapshot/ramses.py:793: UserWarning: Using field 'tform' to distinguish stars. If this is wrong, try editing your config.ini, section [ramses], entry particle-distinguisher.
  warnings.warn("Using field %s to distinguish stars. If this is wrong, try editing your config.ini, section [ramses], entry particle-distinguisher."%pb_name)

In [3]: s.physical_units()                                                                                                                                                                                                

In [4]: s.properties['time']                                                                                                                                                                                              
Out[4]: Unit("-4.09e+00 kpc s km**-1")

In [5]: s.properties['time'].in_units("Gyr")                                                                                                                                                                              
Out[5]: -3.998882056560654

In [6]: s._info['time']                                                                                                                                                                                                   
Out[6]: -1.25693484378412

In [7]: s._info['unit_t']                                                                                                                                                                                                 
Out[7]: 1.00396950379463e+17

In [8]: pynbody.analysis.cosmology.age(s)                                                                                                                                                                                 
Out[8]: 5.382989117242075

In [9]: s.properties                                                                                                                                                                                                      
Out[9]: 
{'a': 0.470006251346598,
 'omegaM0': 0.308600008487701,
 'omegaL0': 0.691399991512299,
 'h': 0.6776999664306641,
 'boxsize': Unit("2.35e+04 kpc"),
 'time': Unit("-4.09e+00 kpc s km**-1")}
```

These negative ages then create very weird interactions with other analysis codes relying on them like Tangos. Although in this use case, the problem arises because we are feeding pynbody corrupted *.txt files, would it be worth re-instating the previous mechanism as a fall-back option, or trying to design a more robust detection mechanism to differentiate between isolated and cosmological runs?

Best,
Martin

**Comment by mkrets** [comment]
Hi @Martin-Rey ,

I agree that we should fall-back on the old method if the namelist is corrupted.

We could do something like this:
```python
except ValueError:
    warnings.warn("Namelist found but unable to read.")
    self._namelist_corrupted = True
```

or fill the namelist as:
```python
    self._namelist["corrupted"] = True
```
Not very elegant solutions.
This is not relevant here but maybe you could elaborate why the files are corrupted by the supercomputer in the ramses repo.

Best,
Michael

**Comment by Martin-Rey** [comment]
Thanks for the quick answer. I think some kind of fall-back would be useful, at least to stay backward compatible as much as possible? Was there a reason to remove the previous implementation? Or edge cases it did not capture?

I actually do not know why my namelist files come out corrupted (i thought it was supercomputer specific, but I checked and they are corrupted on three different machines). This is probably something about my patch, and I will try to track and fix this in parallel (which will likely also address this issue).

**Comment by mkrets** [comment]
The original code included this line:
```python
return self._info['omega_k'] == self._info['omega_l'] == 0
```
which does not capture all cosmologies but the most commonly used LCDM (which is probably > 95% of ramses runs).

I agree that there should be a fall-back to stay backward compatible.

**Comment by Martin-Rey** [comment]
Fixed by #602
